compiler/src/iree/compiler/Dialect/LinalgExt/Utils/Utils.cpp

```diff
@@ -360,4 +360,81 @@ bool isGatherlikeOp(Operation *op) {
   return true;
 }
 
+FailureOr<SmallVector<AffineMap>>
+getIGEMMContractionIndexingMaps(linalg::LinalgOp linalgOp) {
+  MLIRContext *ctx = linalgOp->getContext();
+  return llvm::TypeSwitch<Operation *, FailureOr<SmallVector<AffineMap>>>(
+             linalgOp.getOperation())
+      .Case<linalg::Conv2DNchwFchwOp>(
+          [&](linalg::Conv2DNchwFchwOp convOp) -> SmallVector<AffineMap> {
+            AffineExpr bDim, mDim, nDim0, nDim1, kDim;
+            bindDims(ctx, bDim, mDim, nDim0, nDim1, kDim);
+            auto lhsMap = AffineMap::get(5, 0, {mDim, kDim}, ctx);
+            auto rhsMap = AffineMap::get(5, 0, {bDim, nDim0, nDim1, kDim}, ctx);
+            auto resultMap =
+                AffineMap::get(5, 0, {bDim, mDim, nDim0, nDim1}, ctx);
+            return {lhsMap, rhsMap, resultMap};
+          })
+      .Case<linalg::Conv2DNhwcHwcfOp>(
+          [&](linalg::Conv2DNhwcHwcfOp convOp) -> SmallVector<AffineMap> {
+            AffineExpr bDim, m0Dim, m1Dim, nDim, kDim;
+            bindDims(ctx, bDim, m0Dim, m1Dim, nDim, kDim);
+            auto lhsMap = AffineMap::get(5, 0, {bDim, m0Dim, m1Dim, kDim}, ctx);
+            auto rhsMap = AffineMap::get(5, 0, {kDim, nDim}, ctx);
+            auto resultMap =
+                AffineMap::get(5, 0, {bDim, m0Dim, m1Dim, nDim}, ctx);
+            return {lhsMap, rhsMap, resultMap};
+          })
+      .Default([](Operation *) { return failure(); });
+}
+
+FailureOr<SmallVector<int64_t>> getIGEMMLoopBounds(linalg::LinalgOp linalgOp) {
+  if (!linalg::isaConvolutionOpInterface(linalgOp)) {
+    return failure();
+  }
+  auto filterType = cast<RankedTensorType>(linalgOp->getOperandTypes()[1]);
+  auto accType = cast<RankedTensorType>(linalgOp->getResultTypes()[0]);
```

Comment:
nit: since we're type switching on named conv ops anyway, move these type getters inside the TypeSwitch and drop the `isaConvolutionOpInterface`